sdk/src/wallet/core/operations/background_syncing.rs

```diff
@@ -80,7 +86,7 @@ where
             return Ok(());
         }
         // send stop request
-        self.background_syncing_status.store(2, Ordering::Relaxed);
+        self.request_stop_background_syncing();
         // wait until it stopped
         while self.background_syncing_status.load(Ordering::Relaxed) != 0 {
```

Comment:
Upon seeing this code again, it strikes me that this is a really error-prone way to do this, and that we should be using a different sync type here that we can `await` instead of just sleeping.

Reply:
I got you https://github.com/iotaledger/iota-sdk/pull/1797  😉

Reply:
Ah, good!